### ISSUE TITLE
Licensing of patents

### ISSUE THREAD

**Issue by smathermather** [ContributorCovenant/hippocratic-license#18]
GPLv3, Apache, and other licenses have provisions for licensing of patents. While the hippocratic-license could force re-implementation of software, use of upstream software patents could provide additional constraints for projects.  If the hippocratic-license allowed for licensing of software patents as part of it's text, then re-implementation would be even more difficult.

I am not myself much a fan of software patents, but just thinking though upstream constraints to misuse of code and ideas.

**Comment by paulyc** [comment]
Yeah, ok, being a little facetious here, but I'd say the license should be immediately and irrevocably revoked for all software for any party attempting to enforce any software patent, since  software patents are inherently unethical. (Not that I think filing for a software patent is necessarily unethical, because you have to protect yourself from the patent trolls, but it's a necessarily evil of the United States' broken IP law system, it's the enforcement that takes you into unethical territory). Because most software patents are basically, "doing this thing, but with a computer now!" and otherwise so vague as to apply to every piece of software ever written.

But, hey who am I to say, that is not my final answer, subject to change, it's an interesting topic of discussion anyway.

**Comment by paulyc** [comment]
Yeah it's an interesting question. Won't jump to any conclusions too quick but, the enforcement, rather than the filing, of software patents is usually taken in bad faith, and may be inherently unethical. Unfortunately the filing part is a necessary evil of the broken system of IP law in the United States, which for anything involving computers is a total mess because Congress never actually revised the vast majority of copyright and patent law to account for software, so it's all been up to individual judges to establish law. So it ends up different in every jurisidiction and broken in many. Pretty sure even US software patents aren't actually coded in law, they only exist because a judge or judges said they do, and they could disappear tomorrow if Congress decided to have an opinion and do their jobs for once, or another more important set of judges said so.
